src/core/ddsc/src/dds_writer.c

```diff
@@ -284,15 +284,12 @@ const struct dds_entity_deriver dds_entity_deriver_writer = {
 
 #ifdef DDS_HAS_SHM
 #define DDS_WRITER_QOS_CHECK_FIELDS (QP_LIVELINESS|QP_DEADLINE|QP_RELIABILITY|QP_DURABILITY|QP_HISTORY)
-static bool dds_writer_support_shm(const struct ddsi_config* cfg, const dds_qos_t* qos, const struct dds_topic *tp)
+static bool dds_writer_support_shm(const struct ddsi_config* cfg, const dds_qos_t* qos)
 {
   if (NULL == cfg ||
       false == cfg->enable_shm)
     return false;
 
-  if (!tp->m_stype->fixed_size)
```

Comment:
Arguably, the condition for supporting transfer via Iceoryx is now `tp->m_stype->fixed_size || tp->m_stype->ops->get_serialized_size` (and the same for `dds_reader_supports_shm`).

@MatthiasKillat sorry for agreeing earlier that the topic is no longer needed ... although one could also argue that all `sertype` implementations simply must support `get_serialized_size` and `serialize_into`, that kinda doesn't work too well if the `plist` case then starts out with not supporting it.

Reply:
I agree that this is better to check. I think you mean `&&` in the condition though.